models/location_helpers_test.go

```diff
@@ -0,0 +1,213 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package models_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/reform.v1"
+	"gopkg.in/reform.v1/dialects/postgresql"
+
+	"github.com/percona/pmm-managed/models"
+	"github.com/percona/pmm-managed/utils/testdb"
+)
+
+func TestBackupLocations(t *testing.T) {
+	sqlDB := testdb.Open(t, models.SkipFixtures, nil)
+	defer func() {
+		require.NoError(t, sqlDB.Close())
+	}()
+	db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+
+	t.Run("create", func(t *testing.T) {
+		tx, err := db.Begin()
+		require.NoError(t, err)
+		defer func() {
+			require.NoError(t, tx.Rollback())
+		}()
+
+		q := tx.Querier
+
+		params := models.CreateBackupLocationParams{
+			Name:        "some name",
+			Description: "some desc",
+			FSConfig: &models.FSLocationConfig{
+				Path: "/tmp",
+			},
+		}
+
+		location, err := models.CreateBackupLocation(q, params)
+		require.NoError(t, err)
+		assert.Equal(t, models.FSBackupLocationType, location.Type)
+		assert.Equal(t, params.Name, location.Name)
+		assert.Equal(t, params.Description, location.Description)
+		assert.Equal(t, params.FSConfig.Path, location.FSConfig.Path)
+		assert.NotEmpty(t, location.ID)
+	})
+
+	t.Run("list", func(t *testing.T) {
+		tx, err := db.Begin()
+		require.NoError(t, err)
+		defer func() {
+			require.NoError(t, tx.Rollback())
+		}()
+
+		q := tx.Querier
+
+		params1 := models.CreateBackupLocationParams{
+			Name:        "some name",
+			Description: "some desc",
+			FSConfig: &models.FSLocationConfig{
+				Path: "/tmp",
+			},
+		}
+		params2 := models.CreateBackupLocationParams{
+			Name:        "some name2",
+			Description: "some desc",
+			FSConfig: &models.FSLocationConfig{
+				Path: "/tmp",
+			},
+		}
+
+		loc1, err := models.CreateBackupLocation(q, params1)
+		require.NoError(t, err)
+		loc2, err := models.CreateBackupLocation(q, params2)
+		require.NoError(t, err)
+
+		actual, err := models.FindBackupLocations(q)
+		require.NoError(t, err)
+		var found1, found2 bool
+		for _, channel := range actual {
+			if channel.ID == loc1.ID {
+				found1 = true
+			}
+			if channel.ID == loc2.ID {
+				found2 = true
+			}
+		}
+
+		assert.True(t, found1, "Fist location not found")
+		assert.True(t, found2, "Second location not found")
+	})
+}
+
+func TestBackupLocationValidation(t *testing.T) {
+	sqlDB := testdb.Open(t, models.SkipFixtures, nil)
+	defer func() {
+		require.NoError(t, sqlDB.Close())
+	}()
+	db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+
+	tests := []struct {
+		name     string
+		location models.CreateBackupLocationParams
+		errorMsg string
+	}{
+		{
```

Comment:
can we add two more tests:
1. with both configs are passed
2. no config is passed